contracts/0.8.25/vaults/VaultHub.sol

```diff
@@ -461,59 +473,119 @@ contract VaultHub is PausableUntilWithRoles {
+            reportTotalValue: _reportTotalValue,
+            reportInOutDelta: _reportInOutDelta,
+            reportAccumulatedTreasuryFees: _reportAccumulatedTreasuryFees,
+            reportLiabilityShares: _reportLiabilityShares,
+            reportSlashingReserve: _reportSlashingReserve
+        });
+    }
 
-            IStakingVault vault_ = IStakingVault(_vault);
-            if (!_isVaultHealthy(connection, record) && !vault_.beaconChainDepositsPaused()) {
-                vault_.pauseBeaconChainDeposits();
-            }
+    /// @notice Transfer the bad debt from the donor vault to the acceptor vault
+    /// @param _vault address of the vault that has the bad debt
+    /// @param _vaultAcceptor address of the vault that will accept the bad debt or 0 if the bad debt is socialized to the protocol
+    /// @param _maxSharesToSocialize maximum amount of shares to socialize
+    /// @dev msg.sender must have SOCIALIZE_BAD_DEBT_ROLE
+    function socializeBadDebt(
+        address _vault,
+        address _vaultAcceptor,
+        uint256 _maxSharesToSocialize
+    ) external onlyRole(SOCIALIZE_BAD_DEBT_ROLE) {
+        _requireNotZero(_vault);
+        _requireNotZero(_maxSharesToSocialize);
+
+        VaultConnection storage connection = _vaultConnection(_vault);
+        _requireConnected(connection, _vault); // require connected but may be pending
 
-            emit VaultReportApplied({
-                vault: _vault,
-                reportTimestamp: _reportTimestamp,
-                reportTotalValue: _reportTotalValue,
-                reportInOutDelta: _reportInOutDelta,
-                reportAccumulatedTreasuryFees: _reportAccumulatedTreasuryFees,
-                reportLiabilityShares: _reportLiabilityShares
-            });
+        VaultRecord storage record = _vaultRecord(_vault);
+        uint256 liabilityShares_ = record.liabilityShares;
+        uint256 totalValueShares = _getSharesByPooledEth(_totalValue(record));
+        if (totalValueShares > liabilityShares_) {
+            revert NoBadDebtToSocialize(_vault, totalValueShares, liabilityShares_);
         }
+
+        uint256 badDebtToSocialize = Math256.min(liabilityShares_ - totalValueShares, _maxSharesToSocialize);
+
+        _decreaseLiability(_vault, record, badDebtToSocialize);
+
+        if (_vaultAcceptor == address(0)) {
+            // internalize the bad debt to the protocol
+            _storage().badDebtToInternalize += badDebtToSocialize;
+
+            // disconnect the vault from the hub ?? or ban it ?? or change the owner ??
+        } else {
+            if (_nodeOperator(_vaultAcceptor) != _nodeOperator(_vault)) revert BadDebtSocializationNotAllowed();
+
+            VaultConnection storage connectionAcceptor = _vaultConnection(_vaultAcceptor);
+            _requireConnected(connectionAcceptor, _vaultAcceptor);
+
+            VaultRecord storage recordAcceptor = _vaultRecord(_vaultAcceptor);
+            _increaseLiability(
```

Comment:
:warning: we don't check `liabilityShares == 0` upon disconnect finalization
should do either:
1) forbid pending disconnect acceptors
2) enforce liabilityShares == 0 before final disconnect 

the level can be abused by using vaults without slashing reserves :thinking: